CanHazFunny/CanHazFunny.Tests/JesterTests.cs

```diff
@@ -1,10 +1,86 @@
+﻿using System;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
 
 namespace CanHazFunny.Tests
 {
     [TestClass]
     public class JesterTests
     {
-        
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void JesterNewInstance_GivenNullJokeService_ThrowArgumentNullException()
+        {
+            // Arrange
+            // Act
+            Jester jester = new Jester(null!, new JokePrinter());
+            // Assert
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void JesterNewInstance_GivenNullJokePrinter_ThrowArgumentNullException()
+        {
+            // Arrange
+            // Act
+            Jester jester = new Jester(new JokeService(), null!);
+            // Assert
+        }
+
+        [TestMethod]
+        public void Jester_ImplementIJokeServiceAndJokePrinter_WorkingJokeServiceAndJokePrinter()
+        {
+            // Arrange
+            IJokeService iJokeService = new JokeService();
+            IJokePrinter iJokePrinter = new JokePrinter();
+
+            // Act
+            Jester jester = new Jester(iJokeService, iJokePrinter);
+
+            // Assert
+            Assert.AreEqual(iJokeService, jester.jokeService);
+            Assert.AreEqual(iJokePrinter, jester.jokePrinter);
+        }
+
+        [TestMethod]
+        public void TellJoke_GivenValidJoke_ExcecutesWithoutException()
+        {
+            //Arrange
+            string joke = "Some bad dad joke that makes you roll your eyes.";
+            Mock<IJokeService> mockJokeService = new Mock<IJokeService>();
+            mockJokeService.SetupSequence(JokeService => JokeService.GetJoke())
+                .Returns(joke);
+
+            Mock<IJokePrinter> mockJokePrinter = new Mock<IJokePrinter>();
+            mockJokePrinter.SetupSequence(JokePrinter => JokePrinter.PrintJoke(joke));
```

Comment:
We will get to this later in the class when we cover lambdas, but we typically name the labdam argument with camelCase
```suggestion
            mockJokeService.SetupSequence(jokeService => jokeService.GetJoke())
                .Returns(joke);

            Mock<IJokePrinter> mockJokePrinter = new Mock<IJokePrinter>();
            mockJokePrinter.SetupSequence(jokePrinter => jokePrinter.PrintJoke(joke));
```